keys/keys.go

```diff
@@ -70,6 +70,54 @@ func MakeTableMetadataKey(namespaceID uint32, tableName string) proto.Key {
 	return k
 }
 
+// indexKeyBufferWidth returns a likely cap on the width of the index key.
+// The buffer width can likely accomodate the encoded constant prefix, tableID,
+// indexID, and column values.
+//
+// This cap is inaccurate because the size of the encoding varies, depending
+// on the ints and the bytes being encoded. We really don't care, as long as
+// it provides a cap that prevents the append() builtin, used to populate the
+// buffer, from frequently reallocating more space.
+func indexKeyMaxBufferWidth(columnValues ...[]byte) (width int) {
+	// Accomodate the constant prefix, tableID, and indexID. 10 bytes
+	// for each ID.
+	width += len(TableDataPrefix) + 20
```

Comment:
`s/20/maxVarintSize/g` (you'll need to rebase to master to get `maxVarintSize`).